harness/determined/cli/agent.py

```diff
@@ -79,44 +79,60 @@ def list_slots(args: argparse.Namespace) -> None:
         if r["container_id"]
     }
 
-    def get_task_name(containers: Dict[str, Any], slot: Dict[str, Any]) -> str:
-        if not slot["container"]:
+    def device_type_string(deviceType: devicev1Type) -> str:
+        if deviceType == devicev1Type.TYPE_CUDA:
+            return "cuda"
+        if deviceType == devicev1Type.TYPE_ROCM:
+            return "rocm"
+        if deviceType == devicev1Type.TYPE_CPU:
+            return "cpu"
+        return "unknown"
+
+    def get_task_name(containers: Dict[str, Any], slot: v1Slot) -> str:
+        if not slot.container:
             return "FREE"
 
-        container_id = slot["container"]["id"]
+        container_id = slot.container.id
 
-        if slot["container"] and container_id in containers:
+        if slot.container and container_id in containers:
             return str(containers[container_id]["name"])
 
-        if slot["container"] and (
+        if slot.container and (
             "determined-master-deployment" in container_id
             or "determined-db-deployment" in container_id
         ):
             return f"Determined System Task: {container_id}"
 
+        if slot.container and ("cluster" in container_id):
```

Comment:
not following this change, is there a corresponding ee pr to set container this way in `addSlotToAgent`

Reply:
or, why is this not `if slot.container and slot.container.id == ""`?

Reply:
EE change here https://github.com/determined-ai/determined-ee/pull/743/commits/8d0a2fccafd8b8cf7cc9499118c3a5f9ffdc274d

Reply:
> or, why is this not if slot.container and slot.container.id == ""?

thoughts?

Reply:
To follow the pattern at lines 101-103?

Reply:
That pattern is look for specific containers in kubernetes that we know the name of (the containers for the determined master and db). I guess this makes sense, so that `""` doesn't need to imply "is a determined container", but maybe we can use something more specific than "cluster" - like "dispatcherrm-inuse-slot-placeholder" to be really clear what it means.